src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -66,6 +69,13 @@ public function process(ContainerBuilder $container)
             foreach ($events as $event) {
                 $priority = isset($event['priority']) ? $event['priority'] : 0;
 
+                if (!isset($event['event']) && null !== ($class = $container->getDefinition($id)->getClass()) && ($r = $container->getReflectionClass($class, false)) && $r->hasMethod($event['method'] = $event['method'] ?? '__invoke') && 0 < ($m = $r->getMethod($event['method']))->getNumberOfParameters()) {
+                    $type = $m->getParameters()[0]->getType();
+                    if ($type && !$type->isBuiltin() && Event::class !== $type->getName() && LegacyEvent::class !== $type->getName() && GenericEvent::class !== $type->getName()) {
```

Comment:
Why are all built-in event classes in this guessing omitted?

Reply:
I'm assuming that if someone uses a generic type declaration like that, they have most likely made a mistake. Symfony does not alias the contracts' `Event` class to any real event and imho userland code shouldn't do that either.

Reply:
That's true for `Event` class, but I don't think using the `GenericEvent` class to dispatch events would be a mistake.

Reply:
Using an instance of `GenericEvent` as the event object is of course not a mistake, but using `Symfony\Component\EventDispatcher\GenericEvent` as event name for a domain event is a very bad idea. But I understand that blacklisting it might be a bit opinionated. I'll remove it.